### DISCUSSION TITLE
Where are the definitions like A0E0L-2I0M-1H0T2D0 used?

### DISCUSSION THREAD

**Discussion by mgaldzic** [qudt/qudt-public-repo#138] [Q&A] [answered]
I get how A0E0L-2I0M-1H0T2D0 maps to something like \\(L^-2 M^-1 T^2\\)  for humans. But are these referenced anywhere else? What is the purpose of the forcing the specification of the zero when this attribute is absent?

In essence why include the nulls A0E0, etc? Is there a piece of code somewhere that depends on this encoding?

**Comment by steveraysteveray** [comment, marked as answer]
The quantity kind dimension vectors are referred to by all quantity kinds (or will be when we finish populating), and are used to specify the dimension of every quantity kind. Don't be too hung up on the actual qnames, they are generated to be unique. In a SPARQL query, you can find all commensurate units via the path:
Unit -> Quantity Kind -> Quantity Kind Dimension Vector, and then follow the links back. This is used when doing unit conversion, for example.

Does this start to answer your question?

**Comment by jhodgesatmb** [comment]
The only way to consistently represent dimension qnames is with the full vector. Anyway, this is just the qname and the internal representation is by the individual property and exponent. Check it out.

Jack

**Comment by steveraysteveray** [comment]
@mgaldzic, you might also want to check out our [user guide](https://github.com/qudt/qudt-public-repo/wiki/User-Guide-for-QUDT#3-introducing-dimension-vectors-for-si-units) that talks about this.
